dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +65,267 @@ def __init__(
+            columns (TTableSchemaColumns): The schema columns to match.
+
+        Returns:
+            Tuple[sqlglot.exp.Select, bool]: The outer SELECT statement and a flag indicating if reordering is needed.
+        """
+        if len(parsed_select.selects) == 1 and isinstance(
+            parsed_select.selects[0], sqlglot.exp.Star
+        ):
+            raise ValueError(
+                "\n\nA `SELECT *` was detected in the model query:\n\n"
+                f"{parsed_select.sql(select_dialect)}\n\n"
+                "Model queries using a star (`*`) expression cannot be normalized. "
+                "Please rewrite the query to explicitly specify the columns to be selected.\n"
+            )
+
+        # Wrap parsed select in a subquery
+        subquery = parsed_select.subquery(alias=DLT_SUBQUERY_NAME)
+
+        # Build outer SELECT list
+        selected_columns = []
+        outer_selects: List[sqlglot.exp.Expression] = []
+        for select in parsed_select.selects:
+            if isinstance(select, sqlglot.exp.Alias):
+                name = select.alias
+
+                # NOTE: for bigquery, quoted identifiers are treated as sqlglot.exp.Dot
+            elif isinstance(select, sqlglot.exp.Column) or isinstance(select, sqlglot.exp.Dot):
+                name = select.output_name or select.name
+
+            else:
+                raise ValueError(
+                    "\n\nUnsupported SELECT expression in the model query:\n\n "
+                    f" {select.sql(select_dialect)}\n\nOnly simple column selections like `column`"
+                    " or `column AS alias` are currently supported.\n"
+                )
+
+            norm_casefolded = self._normalize_casefold(name)
+            selected_columns.append(norm_casefolded)
+            outer_selects.append(sqlglot.column(name, table=DLT_SUBQUERY_NAME).as_(norm_casefolded))
+
+        needs_reordering = selected_columns != list(columns.keys())
+
+        # Create the outer select statement
+        outer_select = sqlglot.select(*outer_selects).from_(subquery)
+
+        return outer_select, needs_reordering
+
+    def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]:
+        with self.normalize_storage.extracted_packages.storage.open_file(
+            extracted_items_file, "r"
+        ) as f:
+            select_dialect, select_statement = read_dialect_and_sql(
+                file_obj=f,
+                fallback_dialect=self.config.destination_capabilities.sqlglot_dialect,  # caps are available at this point
+            )
+
+        parsed_select = sqlglot.parse_one(select_statement, read=select_dialect)
+
+        # The query is ensured to be a select statement upstream,
+        # but we double check here
+        if not isinstance(parsed_select, sqlglot.exp.Select):
+            raise ValueError("Only SELECT statements should reach the model normalizer.")
```

Comment:
This error will never come if users to our transformations, but only if they manually yield models as you do in your tests. It would be better to have an error message that reflects this. Something like "The SqlModel you have supplied does not seem to be a SELECT statement." or something like that.